src/book/03-guides/01-main-codebase/07-onboarding-task.mdx

```diff
@@ -0,0 +1,486 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Onboarding Task
+description: Build a simple NUClear program to get comfortable with Git and the NUbots codebase.
+slug: /guides/main/onboarding
+authors:
+  - Thomas O'Brien (@Tom0Brien)
+  - Johanne Montano (@JohanneMontano)
+  - Dexter Konijn (@dex-k)
+---
+
+The onboarding task aims to teach new members about [NUClear](https://nuclear.readthedocs.io/en/latest/). New members will be asked to engage in coding with C++. Prerequisites such as Git, VS code, etc. must be set up first before moving forward with this task. See the [Getting Started](guides/main/getting-started#page-content) page for more details.
+
+## Terminology
+
+### Role
+
+A **Role** is a file that defines the desired purpose or functionality of the binary/program we want to compile. A role file lists what modules we want included in our program/binary.
+
+This is an example of a role file:
+
+```make
+nuclear_role(
+  extension::FileWatcher
+  support::logging::ConsoleLogHandler
+  actuation::KinematicsConfiguration
+  platform::${SUBCONTROLLER}::HardwareIO
+  input::SensorFilter
+  extension::Director
+  actuation::Kinematics
+  actuation::Servos
+  onboarding::Ping
+  onboarding::Pong
+  onboarding::Judge
+)
+```
+
+### Messages
+
+A message is called a neutron. It is essentially a `.proto` file. When a role is built, the proto file becomes imported as a `.hpp` file.
```

Comment:
```suggestion
A message is called a neutron. It is essentially a `.proto` file. The proto file is compiled into C++,  and so it is imported as a `.hpp` C++ header file.
```
I don't know if this is too much info but the existing words seemed to dance around this and didn't make much sense to me.

Reply:
or alternatively 
```suggestion
A message is called a neutron. It is essentially a `.proto` file. When a role is built, the proto file is compiled into C++,  and so it is imported as a `.hpp` C++ header file.
```